### PR TITLE
Fixes for building with PGI compiler

### LINKED ISSUE #262: Build issues with PGI (v19.4)
There are number of issues while building with PGI compiler. List here relevant errors and fix the same.

* -A is injected by PGI: https://www.pgroup.com/userforum/viewtopic.php?f=4&t=7565
* fmt doesn't support PGI : https://github.com/fmtlib/fmt/pull/1581
* Eigen assembly intrinsics : https://listengine.tuxfamily.org/lists.tuxfamily.org/eigen/2010/02/msg00136.html

### PR DESCRIPTION
**Pull request by pramodk**
Supersede #263

- [x] write summary of changes / incompatibilities found, see #262
- [x] Update ticket #262 
- [x] Test with PGI 19.4 as well as 19.10
- [x] Update CMake to automatically inject necessary PGI flags : 

```
module load unstable gcc/8.3.0 pgi cuda intel-mpi cmake python-dev

cmake .. -DCMAKE_INSTALL_PREFIX=./install \
-DFLEX_EXECUTABLE=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/flex-2.6.3-3fjqz4ot4m/bin/flex \
-DBISON_EXECUTABLE=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/bison-3.0.5-xs5kockuyz/bin/bison \
-DPYTHON_EXECUTABLE=$(which python) \
-DNMODL_FORMATTING:BOOL=ON \
-DClangFormat_EXECUTABLE=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/compilers/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/llvm-9.0.0-63wiwhr7ki/bin/clang-format \
-DCMAKE_INSTALL_PREFIX=`pwd`/install
```

<details><summary>Old Build Command</summary>
<p>
cmake .. -DFLEX_EXECUTABLE=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/flex-2.6.3-3fjqz4ot4m/bin/flex\
 -DBISON_EXECUTABLE=/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/tools/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/bison-3.0.5-xs5kockuyz/bin/bison\
 -DPYTHON_EXECUTABLE=$(which python)\
 -DCMAKE_BUILD_TYPE=Release\
 -DNMODL_FORMATTING:BOOL=ON\
 -DCMAKE_CXX_FLAGS=" -DJSON_SKIP_UNSUPPORTED_COMPILER_CHECK=1 --diag_suppress 611,997,82,111,186,1625,1097,177 --display_error_number -DEIGEN_DONT_VECTORIZE=1"\
 -DCMAKE_BUILD_TYPE=Custom
</p>
</details>

fixes #262

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -12,7 +12,7 @@
 	url = https://github.com/eigenteam/eigen-git-mirror.git
 [submodule "ext/spdlog"]
 	path = ext/spdlog
-	url = https://github.com/gabime/spdlog.git
+	url = https://github.com/pramodk/spdlog.git
 [submodule "ext/fmt"]
 	path = ext/fmt
-	url = https://github.com/fmtlib/fmt.git
+	url = https://github.com/pramodk/fmt.git
```
Comment (pramodk) [inline .gitmodules]: Have to see this : PR upstream is pending. So most likely we have to fork under BlueBrain.

File: cmake/CompilerHelper.cmake [hunk 2]
```diff
@@ -5,3 +5,22 @@ if("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
...
   endif()
 endif()
+
+if(CMAKE_CXX_COMPILER_ID MATCHES "PGI")
+  # CMake adds standard complaint PGI flag "-A" which breaks compilation of of spdlog and fmt
+  set(CMAKE_CXX11_STANDARD_COMPILE_OPTION --c++11)
+  set(CMAKE_CXX14_STANDARD_COMPILE_OPTION --c++14)
+
+  # PGI with llvm code generation doesn't have necessary assembly intrinsic headers
+  add_compile_definitions(EIGEN_DONT_VECTORIZE=1)
```
Comment (ohm314) [inline cmake/CompilerHelper.cmake]: I guess it's ok, but I find this an astonishing place to put this command - I would move this into the main `CMakeLists.txt` to where we discover the Eigen package.

File: cmake/CompilerHelper.cmake [hunk 3]
```diff
@@ -5,3 +5,22 @@ if("${CMAKE_CXX_COMPILER_ID}" MATCHES "GNU")
...
+
+  # nlohmann/json doesn't check for PGI compiler
+  add_compile_definitions(JSON_SKIP_UNSUPPORTED_COMPILER_CHECK=1)
```
Comment (ohm314) [inline cmake/CompilerHelper.cmake]: dito

File: cmake/CompilerHelper.cmake (hunk 2 above)
Comment (pramodk) [reply cmake/CompilerHelper.cmake]: yeah agree!

File: cmake/CompilerHelper.cmake (hunk 3 above)
Comment (pramodk) [reply cmake/CompilerHelper.cmake]: adressed

**Pull request merged**